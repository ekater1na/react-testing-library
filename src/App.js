import React, { useState } from "react";
import "./App.css";

const Search = ({ value, onChange, children }) => (
  <div>
    <label htmlFor="search">{children}</label>
    <input
      id="search"
      type="text"
      placeholder="search text"
      value={123}
      onChange={onChange}
    />
  </div>
);

const App = () => {
  const [search, setSearch] = useState("");

  const handleChange = ({ target }) => {
    setSearch(target.value);
  };

  return (
    <div>
      <img src="" alt="sun"></img>
      <Search value={search} onChange={handleChange}>
        Search:
      </Search>
      <p>Searches for {search ? search : "..."}</p>
    </div>
  );
};

export default App;
