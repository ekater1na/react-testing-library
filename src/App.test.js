import { render, screen } from "@testing-library/react";
import App from "./App";

describe("App", () => {
  it("renders App component", () => {
    const { asFragment } = render(<App />);
    expect(asFragment(<App />)).toMatchSnapshot();
   
  });

  it("renders Search text", () => {    
    render(<App />);   
    expect(screen.getByText(/search:/i)).toBeInTheDocument();
  });

  
  it("renders buttton", () => {    
    render(<App />);    
    // screen.debug();
    expect(screen.getByRole('textbox')).toBeInTheDocument();
    expect(screen.getByLabelText(/search/i)).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/search text/i)).toBeInTheDocument();
    expect(screen.getByAltText('sun')).toBeInTheDocument();
    expect(screen.getByDisplayValue('123')).toBeInTheDocument();

  });
});
