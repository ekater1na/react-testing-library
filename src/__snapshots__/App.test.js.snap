// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders App component 1`] = `
<DocumentFragment>
  <div>
    <img
      alt="sun"
      src=""
    />
    <div>
      <label
        for="search"
      >
        Search:
      </label>
      <input
        id="search"
        placeholder="search text"
        type="text"
        value="123"
      />
    </div>
    <p>
      Searches for ...
    </p>
  </div>
</DocumentFragment>
`;
